subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyTaskIntegrationSpec.groovy

```diff
@@ -2239,4 +2242,121 @@ class CopyTaskIntegrationSpec extends AbstractIntegrationSpec {
         "rename(Transformer)"       | "rename(org.gradle.internal.Transformers.noOpTransformer())"
     }
 
+    // region duplicates in compressed files
+    def "encountering duplicates in a zipTree vs an unzipped dir with DuplicateStrategy.FAIL should give a clear error"() {
+        given: "a directory with a file"
+        TestFile unzippedDir = file("before/files")
+        unzippedDir.file("sub/c.txt").createFile()
+
+        and: "a zip file containing it"
+        TestFile zipFile = file("before/files.zip")
+        unzippedDir.zipTo(zipFile)
+
+        and: "a copy task that copies from both of these, failing on duplicates"
+        buildFile << """
+            task (copy, type: Copy) {
+                from '${separatorsToUnix(unzippedDir.path)}'
+                from zipTree('${separatorsToUnix(zipFile.path)}')
```

Comment:
I think it's OK to duplicate the file paths here vs having to normalize the paths.  We really only normalize paths like this when they must be absolute in the build script or some output will produce them. 

```suggestion
                from 'before/files'
                from zipTree('before/files.zip')
```